capstone/fitness/src/main/java/com/google/sps/servlets/AbstractAuthServlet.java

```diff
@@ -1,42 +1,153 @@
 package com.google.sps.servlets;
-
+ 
 import com.google.appengine.api.users.UserService;
 import com.google.appengine.api.users.UserServiceFactory;
-
+import com.google.api.client.auth.oauth2.AuthorizationCodeFlow;
+import com.google.api.client.auth.oauth2.Credential;
+import com.google.api.client.util.DateTime;
+import com.google.api.client.extensions.appengine.auth.oauth2.AbstractAppEngineAuthorizationCodeServlet;
+import com.google.api.client.extensions.appengine.http.UrlFetchTransport;
+import com.google.api.client.http.HttpTransport;
+import com.google.api.client.json.JsonFactory;
+import com.google.api.client.json.jackson2.JacksonFactory;
+import com.google.api.services.calendar.Calendar;
+import com.google.api.services.calendar.model.CalendarList;
+import com.google.api.services.calendar.model.CalendarListEntry;
+import com.google.api.services.calendar.model.Event;
+import com.google.api.services.calendar.model.Events;
+import com.google.gson.Gson;
+import com.google.gson.reflect.TypeToken;
+import com.google.sps.util.*;
+import com.google.sps.progress.*;
 import java.io.IOException;
-
-
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 import javax.servlet.ServletException;
+import java.time.format.DateTimeFormatter;  
+import java.time.LocalDateTime;   
+import java.util.*;
 
 
-import com.google.api.client.extensions.appengine.auth.oauth2.AbstractAppEngineAuthorizationCodeServlet;
-import com.google.api.client.auth.oauth2.AuthorizationCodeFlow;
 
-@WebServlet("/abstract")
+// AbstractAuthServlet initializes the OAuth process and does calendar functions. 
+@WebServlet("/auth-servlet")
 public class AbstractAuthServlet extends AbstractAppEngineAuthorizationCodeServlet {
-  String nickname = UserServiceFactory.getUserService().getCurrentUser().getNickname();
 
+  String APPLICATION_NAME = "Marathon App";
```

Comment:
Try making some of the variables private static, eg. colorId and exerciseDuration and dtf